File: akka-benchmark/src/main/java/io/sarl/akka/hierarchy/Parent.java
package io.sarl.akka.hierarchy;

import akka.actor.AbstractActor;
import akka.actor.ActorRef;
import akka.actor.Props;
import akka.event.Logging;
import akka.event.LoggingAdapter;
import io.sarl.akka.sample.Printer.Greeting;

import java.util.ArrayList;
import java.util.List;

public class Parent extends AbstractActor {
    static public Props props() {
        return Props.create(Parent.class, Parent::new);
    }

    static public class Start {
        //
    }

    private static final int CHILDREN_COUNT = 10;

    private List<ActorRef> children = new ArrayList<>();

    public Parent() {
        for(int idx = 0; idx < CHILDREN_COUNT; idx++) {
            children.add(getContext().actorOf(Props.create(Child.class)));
        }
    }

    @Override
    public Receive createReceive() {
        return receiveBuilder()
                .match(Start.class, message -> {
                    for(ActorRef child : children) {
                        child.tell(new Child.PrintSomething(child.path().toString()), getSelf());
                    }
                })
                .build();
    }
}

File: akka-benchmark/src/main/kotlin/io/sarl/akka/hierarchy/Parent.kt
package io.sarl.akka.hierarchy

import akka.actor.AbstractActor
import akka.actor.ActorRef
import akka.actor.Props
import akka.japi.Creator

import java.util.ArrayList

class Parent : AbstractActor() {

    private val children = ArrayList<ActorRef>()

    class Start//

    init {
        for (idx in 0 until CHILDREN_COUNT) {
            children.add(context.actorOf(Props.create(Child::class.java)))
        }
    }

    override fun createReceive(): AbstractActor.Receive {
        return receiveBuilder()
                .match(Start::class.java) { message ->
                    for (child in children) {
                        child.tell(Child.PrintSomething(child.path().toString()), self)
                    }
                }
                .build()
    }

    companion object {
        fun props(): Props {
            return Props.create<Parent>(Parent::class.java) { Parent() }
        }

        private val CHILDREN_COUNT = 10
    }
}
